src/main/java/net/sf/jabref/model/entry/BibEntry.java

```diff
@@ -104,6 +111,58 @@ public BibEntry(String id, String type) {
     }
 
     /**
+     * Returns the text stored in the given field of the given bibtex entry
+     * which belongs to the given database.
+     * <p>
+     * If a database is given, this function will try to resolve any string
+     * references in the field-value.
+     * Also, if a database is given, this function will try to find values for
+     * unset fields in the entry linked by the "crossref" field, if any.
+     *
+     * @param field    The field to return the value of.
+     * @param database maybenull
+     *                 The database of the bibtex entry.
+     * @return The resolved field value or null if not found.
+     */
+    public Optional<String> getResolvedField(String field, BibDatabase database) {
+        Objects.requireNonNull(this, "entry cannot be null");
+
+        if (TYPE_HEADER.equals(field) || OBSOLETE_TYPE_HEADER.equals(field)) {
+            Optional<EntryType> entryType = EntryTypes.getType(getType(), BibDatabaseMode.BIBLATEX);
+            if (entryType.isPresent()) {
+                return Optional.of(entryType.get().getName());
+            } else {
+                return Optional.of(EntryUtil.capitalizeFirst(getType()));
+            }
+        }
+
+        if (KEY_FIELD.equals(field)) {
+            return getCiteKeyOptional();
+        }
+
+        // Changed this to also consider alias fields, which is the expected
+        // behavior for the preview layout and for the check whatever all fields are present.
+        // TODO: But there might be unwanted side-effects?!
+        Optional<String> result = getFieldOrAlias(field);
+
+        // If this field is not set, and the entry has a crossref, try to look up the
+        // field in the referred entry: Do not do this for the bibtex key.
+        if (!result.isPresent() && (database != null)) {
+            Optional<String> crossrefKey = getField(FieldName.CROSSREF);
```

Comment:
Better use a flatmap, it would only get evaluated if a value is present in all fields and addtionally you can add a filter condition for checking !isEmpty 
http://www.nurkiewicz.com/2013/08/optional-in-java-8-cheat-sheet.html (scroll down to end to see the "bigger example")